fibmod.asm

```diff
@@ -0,0 +1,65 @@
+# COMP3410 Program Template
+# Author: Warner Johnson
+# Assignment: PA1
+# Date: 2/4/2015
+#
+# fibonacci.asm
+# Input: User input
+# Output: Print to terminal N Fibonacci numbers
+
+##########################################################
+# .data segment. Constant and variable definitions go here.
+##########################################################
+
+	.data
+	
+prompt: .asciiz "\Please enter an integer:\n"
+delimiter: .asciiz "\, "
+
+###############################################
+# .text segment. Assembly instructions go here.
+###############################################
+	.text
+	      la   $a0, prompt      # load address of prompt into first argument register
+	      li   $v0, 4           # load print string call into first value register
+	      syscall		    # make system call that prints string prompt
+	      
+	      li   $v0, 5	    # load user input integer call into first system register
+	      syscall               # make system call that takes user input
+	      
+	      move $s0, $v0	    #move user input from first system register to save register one
+	      li $s1, 1		    #place one in save register two
+	      li $s2, 1		    #place one in save register three
+	      li $s3, 1	            #place one in save register four
+	      bgtz $s0, printOne    #branch to printOne if save register one has a value greater than zero
+	      
+printOne:     li $a0, 1		    #load one into the first argument register
+	      li $v0, 1		    #load one into the first system register
+	      syscall               #print one
+	      subi $s0, $s0, 1      #subtract one from the first save register
+	      jal delimit
+	      bgtz $s0, loop	    #branch to loop if save register one is still greater than zero
+
```

Comment:
-2 Should always test boundary cases.  If you enter a number < 2 the program still prints the first 2 fibonacci numbers.